### PR TITLE
Reuse <style> nodes in IE11

### PR DESCRIPTION
**Pull request by MLoughry**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Most browsers, according to spec, determine cascading priority for rules with the same specificity based upon the DOM order of the style nodes in the `<head>`. When a new `<style> node is added, it inserts it in the appropriate place in `document.styleSheets`.

Internet Explorer 11 differs (because of course it does), instead determining the cascading order based upon when the `<style>` node was *created*. It merely appends any new stylesheets to the end of the `document.styleSheets` array, regardless of where in the DOM they are inserted.

This change preserves the intended cascading order of Fabric styles by re-using the single `<style>` node in IE11.

#### Focus areas to test

This will affect rendering performance in IE. We should ensure the regression is not too great.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9618)

### CODE REVIEW

**Comment by MLoughry** [comment]
@dzearing, who has background on this change.

File: packages/merge-styles/src/Stylesheet.ts [hunk 1]
```diff
@@ -63,6 +63,11 @@ export interface IStyleSheetConfig {
 }
 
 const STYLESHEET_SETTING = '__stylesheet__';
+/**
+ * MSIE 11 doesn't cascade styles based on DOM ordering, but rather on the order that each style node
+ * is created. As such, to maintain consistent priority, IE11 should reuse a single style node.
+ */
+const REUSE_STYLE_NODE = /rv:11.0/.test(navigator.userAgent);
```
Comment (dzearing) [inline packages/merge-styles/src/Stylesheet.ts]: navigator may not exist in a server rendered environment. You should probably check that.

Comment (MLoughry) [reply packages/merge-styles/src/Stylesheet.ts]: Fixed.
```
PS E:\VSO\office-ui-fabric-react> node -p "typeof navigator !== 'undefined' && /rv:11.0/.test(navigator.userAgent)"
false
```

**Comment by dzearing** [comment]
Will merge once this is validated to fix the issue locally

**Comment by MLoughry** [comment]
@dzearing @kenotron I've confirmed this fixes the issue locally in OWA in IE11.

**Comment by dzearing** [comment]
I believe this change has nothing to do with the a11y break.

**Comment by xugao** [comment]
@MLoughry - this change is giving us a bit trouble to polyfill css variables. also I don't quite understand this issue - I can't seem to repro: https://codepen.io/xugao/pen/rNLeYWv ?

**Pull request merged**